### PR TITLE
Defines Auth multi-tenancy references in index.d.ts.

### PR DESCRIPTION
**Pull request by bojeil-google**
Defines Auth multi-tenancy references in index.d.ts.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -952,6 +960,11 @@ declare namespace admin.auth {
      * The buffer of bytes representing the user’s password salt.
      */
     passwordSalt?: Buffer;
+  
+    /**
+     * The identifier of the tenant where user is to be imported to.
+     * When not provided, the user is uploaded to the default parent project.
```
Comment (wuyanna) [inline src/index.d.ts]: If not provided, the user is uploaded to the tenant specified in the request parameter `tenantId`

File: src/index.d.ts [hunk 2]
```diff
@@ -1051,31 +1064,128 @@ declare namespace admin.auth {
 
   type TenantType = 'lightweight' | 'full_service';
```
Comment (wuyanna) [inline src/index.d.ts]: Do you need to add comment for this as well?

File: src/index.d.ts [hunk 3]
```diff
@@ -1051,31 +1064,128 @@ declare namespace admin.auth {
...
+   * be used instead. When configuring these providers, note that tenants will inherit
+   * whitelisted domains and authenticated redirect URIs of their parent project.
+   *
+   * All other settings of a tenant will also be inherited. These will need to be managed
+   * from the Cloud Console UI.
+   */
   interface Tenant {
+
+    /**
+     * The current tenant identifier.
```
Comment (wuyanna) [inline src/index.d.ts]: Do you think we should be more specific? like `The current tenant identifier to be passed in the requests.`

File: src/index.d.ts (hunk 2 above)
Comment (wuyanna) [reply src/index.d.ts]: nvm, saw you added below

File: src/index.d.ts (hunk 3 above)
Comment (bojeil-google) [reply src/index.d.ts]: For the `Tenant` interface, this is only returned in operations like `getTenant`, `createTenant`, `updateTenant` and `listTenants`. It doesn't seem to apply here. I think it applies more to `TenantAwareAuth#tenantId`. Check below to see the changes I added.

File: src/index.d.ts (hunk 1 above)
Comment (bojeil-google) [reply src/index.d.ts]: I expanded the description. Please take another look.

File: src/index.d.ts [hunk 4]
```diff
@@ -1820,18 +1933,101 @@ declare namespace admin.auth {
...
+     *
+     * @param tenantId The `tenantId` corresponding to the tenant to delete.
+     *
+     * @return An empty promise fulfilled once the tenant has been deleted.
+     */
     deleteTenant(tenantId: string): Promise<void>;
+
+    /** 
+     * Creates a new tenant.
+     * When creating new tenants, tenants that use separare billing and quota will require their
```
Comment (kevinthecheung) [inline src/index.d.ts]: separate

File: src/index.d.ts [hunk 5]
```diff
@@ -602,6 +602,10 @@ declare namespace admin.auth {
      * resets, password or email updates, etc).
      */
     tokensValidAfterTime?: string;
+  
+    /**
+     * The user's tenant identifier if available.
```
Comment (kevinthecheung) [inline src/index.d.ts]: The ID of the tenant the user belongs to

File: src/index.d.ts [hunk 6]
```diff
@@ -727,6 +731,10 @@ declare namespace admin.auth {
        * `"google.com"`, `"twitter.com"`, or `"custom"`.
        */
       sign_in_provider: string;
+
+      /**
+       * The user's tenant ID if available.
```
Comment (kevinthecheung) [inline src/index.d.ts]: same

File: src/index.d.ts [hunk 7]
```diff
@@ -1051,31 +1067,128 @@ declare namespace admin.auth {
...
+   * be used instead. When configuring these providers, note that tenants will inherit
+   * whitelisted domains and authenticated redirect URIs of their parent project.
+   *
+   * All other settings of a tenant will also be inherited. These will need to be managed
+   * from the Cloud Console UI.
+   */
   interface Tenant {
+
+    /**
+     * The current tenant identifier.
```
Comment (kevinthecheung) [inline src/index.d.ts]: Can you omit "current" here and throughout?

File: src/index.d.ts [hunk 8]
```diff
@@ -1051,31 +1067,128 @@ declare namespace admin.auth {
...
+      /**
+       * Whether password is required for email sign-in. When not required,
+       * email sign-in can be performed with password or via email link sign-in.
+       */
       passwordRequired?: boolean;
     };
   }
 
+  /**
+   * Interface representing the properties to set on a new tenant to be created.
```
Comment (kevinthecheung) [inline src/index.d.ts]: ... a new tenant.

File: src/index.d.ts [hunk 9]
```diff
@@ -1820,18 +1933,101 @@ declare namespace admin.auth {
     ): Promise<admin.auth.AuthProviderConfig>;
   }
 
+  /**
+   * Tenant aware `Auth` interface used for managing user, configuring SAML/OIDC providers,
```
Comment (kevinthecheung) [inline src/index.d.ts]: Tenant-aware

Comment (kevinthecheung) [inline src/index.d.ts]: users

**Pull request merged**